integration-tests/src/test/java/com/linkedin/kafka/clients/largemessage/LargeMessageIntegrationTest.java

```diff
@@ -116,6 +117,18 @@ public void onCompletion(RecordMetadata recordMetadata, Exception e) {
         consumptionStarted = true;
       }
       for (ConsumerRecord<String, String> consumerRecord : records) {
+        // Verify headers
+        Map<String, byte[]> headers = LiKafkaClientsTestUtils.fetchSpecialHeaders(consumerRecord.headers());
+        assertTrue(headers.containsKey(Constants.TIMESTAMP));
+        assertEquals(8, headers.get(Constants.TIMESTAMP).length);
+        long eventTimestamp = PrimitiveEncoderDecoder.decodeLong(headers.get(Constants.TIMESTAMP), 0);
+        assertTrue(eventTimestamp >= startTime && eventTimestamp <= System.currentTimeMillis());
```

Comment:
Why not get System.currentTimeMillis() before the for loop and use it?

Reply:
IMO, it just pollutes the test case with more variables.